.azure-pipelines/windows.sh

```diff
@@ -62,27 +62,36 @@ clone_repos
 if [ "$MODEL" == "64" ] ; then
     MAKE_FILE="win64.mak"
     LIBNAME=phobos64.lib
-elif [ "$MODEL" == "32mscoff" ] ; then
+elif [ "$MODEL" == "32" ] ; then
     MAKE_FILE="win64.mak"
     LIBNAME=phobos32mscoff.lib
-else
-    export LIB="$PWD/dmd2/windows/lib"
+else # 32omf
     MAKE_FILE="win32.mak"
     LIBNAME=phobos.lib
 fi
 
 ################################################################################
 # Build DMD (incl. building and running the unittests)
 ################################################################################
-if [ "$MODEL" == "32omf" ] ; then
-    DMD_BIN_PATH="$DMD_DIR/generated/windows/release/32/dmd"
-else
-    DMD_BIN_PATH="$DMD_DIR/generated/windows/release/$MODEL/dmd"
+
+# no `-debug` for unittests build with old host compilers (to avoid compile errors)
+disable_debug_for_unittests=()
+if [[ "$HOST_DMD_VERSION" == "2.079.0" ]]; then
+    disable_debug_for_unittests=(ENABLE_DEBUG=0)
 fi
 
-cd "$DMD_DIR/compiler/src"
-"$DM_MAKE" -f "$MAKE_FILE" MAKE="$DM_MAKE" BUILD=debug unittest
-DFLAGS="-L-LARGEADDRESSAWARE" "$DM_MAKE" -f "$MAKE_FILE" MAKE="$DM_MAKE" GEN="$DMD_DIR\generated" reldmd-asserts
+# avoid the DMC runtime and its limitations for the compiler and {build,run}.d tools themselves
+TOOL_MODEL="$MODEL"
+if [[ "$MODEL" == "32omf" ]]; then
+    TOOL_MODEL=32
+fi
+
+cd "$DMD_DIR"
+"$HOST_DC" -m$TOOL_MODEL compiler/src/build.d -ofgenerated/build.exe
```

Comment:
Initially I haven't bothered with setting a model for the `build.d` compilation. Then noticed this: https://github.com/dlang/dmd/blob/31c5a894d2680c56870a132a02586403fb9e265a/compiler/src/build.d#L135-L140